### PR TITLE
#F StringProviding integration

### PR DESCRIPTION
**Pull request by EgorovEI**
StringProviding is set when configuration is done. Then this StringProviding will be used to retrieve remote string if exists, otherwise localized value is taken from local file.

### CODE REVIEW

File: GliaWidgets/Public/Glia/Glia.Deprecated.swift [hunk 1]
```diff
@@ -114,12 +114,15 @@ extension Glia {
...
+            createdInteractor.withConfiguration { [weak createdInteractor, weak self] in
                 guard let interactor = createdInteractor else { return }
-                    interactor.state = GliaCore.sharedInstance
-                        .getCurrentEngagement()?.engagedOperator
-                        .map(InteractorState.engaged) ?? interactor.state
-                    callback()
+
+                self?.stringProviding = .init(getRemoteString: GliaCore.sharedInstance.localeProvider.getRemoteString)
+
+                interactor.state = GliaCore.sharedInstance
```
Comment (igorkravchenko) [inline GliaWidgets/Public/Glia/Glia.Deprecated.swift:122]: [Same](https://github.com/salemove/ios-sdk-widgets/pull/779/files#r1333081122).

File: GliaWidgets/Public/Glia/Glia.swift [hunk 2]
```diff
@@ -126,10 +126,11 @@ public class Glia {
         interactor = createdInteractor
 
         if let callback = completion {
-            createdInteractor.withConfiguration { [weak createdInteractor] in
+            createdInteractor.withConfiguration { [weak createdInteractor, weak self] in
                 guard let interactor = createdInteractor else { return }
 
                 // TODO: Configure string providing from Core SDK here.
+                self?.stringProviding = .init(getRemoteString: GliaCore.sharedInstance.localeProvider.getRemoteString)
```
Comment (igorkravchenko) [inline GliaWidgets/Public/Glia/Glia.swift:133]: [Same](https://github.com/salemove/ios-sdk-widgets/pull/779/files#r1333081122).

**Review by igorkravchenko — COMMENTED** [review]
Left come comments regarding dependency injection.

File: GliaWidgets/Public/Glia/Glia.Deprecated.swift [hunk 3]
```diff
@@ -114,12 +114,15 @@ extension Glia {
...
         if let callback = completion {
-            createdInteractor.withConfiguration { [weak createdInteractor] in
+            createdInteractor.withConfiguration { [weak createdInteractor, weak self] in
                 guard let interactor = createdInteractor else { return }
-                    interactor.state = GliaCore.sharedInstance
-                        .getCurrentEngagement()?.engagedOperator
-                        .map(InteractorState.engaged) ?? interactor.state
-                    callback()
+
+                self?.stringProviding = .init(getRemoteString: GliaCore.sharedInstance.localeProvider.getRemoteString)
```
[the remark this answers is not in the document]
Comment (EgorovEI) [reply GliaWidgets/Public/Glia/Glia.Deprecated.swift]: Fixed 20346c66f050980bcdc2774225a81282b714a7c7

File: GliaWidgets/Public/Glia/Glia.Deprecated.swift (hunk 1 above)
Comment (EgorovEI) [reply GliaWidgets/Public/Glia/Glia.Deprecated.swift]: Fixed 20346c66f050980bcdc2774225a81282b714a7c7

Comment (EgorovEI) [reply GliaWidgets/Public/Glia/Glia.Deprecated.swift]: Now it's fixed 😄 2550ea25f4220bad6f2283d72d39110572d42692

File: GliaWidgets/Public/Glia/Glia.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/Public/Glia/Glia.swift]: Now it's fixed 😄 2550ea25f4220bad6f2283d72d39110572d42692

**Review by igorkravchenko — APPROVED** [review]

**Review by gersonnoboa — APPROVED** [review]
Seems great, let's see what others say but it looks like the Core SDK interface is good.

**Review by gersonnoboa — APPROVED** [review]

**Comment by EgorovEI** [comment]
!squash

**Pull request merged**